src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -112,13 +117,21 @@ class SnowflakeStreamingIngestChannelInternal implements SnowflakeStreamingInges
                     String.format("%s_%s", name, channelSequencer),
                     0,
                     this.owningClient.getAllocator().getLimit());
-    this.arrowBuffer = new ArrowRowBuffer(this);
+    this.rowBuffer = createRowBuffer(client);
     this.encryptionKey = encryptionKey;
     this.encryptionKeyId = encryptionKeyId;
     this.onErrorOption = onErrorOption;
     logger.logInfo("Channel={} created for table={}", this.channelName, this.tableName);
   }
 
+  private RowBuffer<T> createRowBuffer(SnowflakeStreamingIngestClientInternal<T> client) {
+    // TODO: The circular dependency SnowflakeStreamingIngestClientInternal <-> RowBuffer
```

Comment:
Not sure if I understand, why the `SnowflakeStreamingIngestClientInternal` type is depend on `RowBuffer`? i don't see a circular dependency here

Reply:
sorry, meant `SnowflakeStreamingIngestChannelInternal`, changed